### PR TITLE
reports: include number nodes actively scanned

### PR DESCRIPTION
**Pull request by levihackerman-102**
I tried these changes but it's giving an OGNL error so they are most likely incorrect. Need help in resolving them :(
I also forgot to sign the commits. I'll rebase and fix them later.

Part of zaproxy/zaproxy#7022.

### CODE REVIEW

**Comment by levihackerman-102** [comment]
@kingthorin @psiinon @thc202 Can someone explain why my code changes are giving an OGNL error when I try to generate a report after an active scan?

**Comment by kingthorin** [comment]
Could you quote the full error and attach the session (zip it up).

**Comment by levihackerman-102** [comment]
> Could you quote the full error and attach the session (zip it up).
Screenshot from 2022-03-20 03-18-55
[Anshul_zapsession.zip](https://github.com/zaproxy/zap-extensions/files/8310207/Anshul_zapsession.zip)

**Comment by thc202** [comment]
Did you install the automation add-on? The exception stack trace would provide more info.

**Comment by kingthorin** [comment]
What version of Java are you using?

**Comment by levihackerman-102** [comment]
> What version of Java are you using?

OpenJDK 11

**Comment by levihackerman-102** [comment]
> Did you install the automation add-on? The exception stack trace would provide more info.

I think I installed all the add-ons. How do I view the exception stack trace?

**Comment by thc202** [comment]
It should be in the log file and the Output tab if you are running from source (or in dev mode).

**Comment by kingthorin** [comment]
The code also needs a spotlessApply

**Comment by kingthorin** [comment]
Thanks! Life happens, no problem. Just checking in :grinning:

**Comment by kingthorin** [comment]
@levihackerman-102 any news/plans for this?

**Comment by kingthorin** [comment]
@levihackerman-102 any news/plans for this?

**Comment by thc202** [comment]
Should be ready for review.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java [hunk 1]
```diff
@@ -56,6 +59,7 @@ public ActiveScanJobResultData(String jobName, ActiveScan activeScan) {
                         plugin.getTimeStarted().getTime() - plugin.getTimeFinished().getTime());
                 data.setStrength(plugin.getAttackStrength());
                 data.setThreshold(plugin.getAlertThreshold());
+                data.setThreshold(plugin.getAlertThreshold());
```
Comment (kingthorin) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java:62]: Why twice?

File: addOns/reports/src/main/zapHomeFiles/reports/traditional-html-plus/report.html [hunk 2]
```diff
@@ -190,6 +190,11 @@ <h3 th:text="#{report.template.pass.list}">Passing Rules</h3>
 	</th:block>
 
 	<h2 th:text="#{report.sites.title}">Sites</h2>
+
+	<strong>Number of Sites tree nodes actively scanned:</strong>
+	<th:block
+		th:text="${reportData.reportObjects.get('activeScanData') != null ? reportData.reportObjects.get('activeScanData').numberScannedNodes : 0}">42</th:block>
```
Comment (kingthorin) [inline addOns/reports/src/main/zapHomeFiles/reports/traditional-html-plus/report.html:196]: 42 is just an example?

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java (hunk 1 above)
Comment (thc202) [reply addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java:62]: Mistake reverting changes...

File: addOns/reports/src/main/zapHomeFiles/reports/traditional-html-plus/report.html (hunk 2 above)
Comment (thc202) [reply addOns/reports/src/main/zapHomeFiles/reports/traditional-html-plus/report.html:196]: Yes, that's just placeholder value.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java (hunk 1 above)
Comment (thc202) [reply addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java]: Removed.

**Pull request merged**